drivers/wifi/nrf700x/zephyr/src/qspi/src/nrf700x_shell.c

```diff
@@ -0,0 +1,885 @@
+		qdev->write(addr, buff, test_chunk * 4);
+
+		(hl_flag) ? qdev->hl_read(addr, rxbuff, test_chunk * 4) :
+			    qdev->read(addr, rxbuff, test_chunk * 4);
+
+		if (memcmp(buff, rxbuff, test_chunk * 4) != 0) {
+			shell_print(shell, "memtest failed");
+			k_free(rxbuff);
+			k_free(buff);
+			return -1;
+		}
+		rem_words -= 2000;
+		chunk_no++;
+	}
+	shell_print(shell, "memtest PASSED");
+	k_free(rxbuff);
+	k_free(buff);
+
+	return 0;
+}
+
+void get_sleep_stats(uint32_t addr, uint32_t *buff, uint32_t wrd_len)
+{
+#if CONFIG_NRF700X_ON_QSPI
+	qspi_cmd_wakeup_rpu(&qspi_perip, 0x1);
+	LOG_DBG("Waiting for RPU awake...\n");
+
+	qspi_validate_rpu_wake_writecmd(&qspi_perip);
+	LOG_DBG("exited qspi_validate_rpu_wake_writecmd(). Waiting for rpu_awake.....\n");
+
+	qspi_wait_while_firmware_awake(&qspi_perip);
+	LOG_DBG("exited qspi_wait_while_rpu_awake()...\n");
+
+	(hl_flag) ? qdev->hl_read(addr, buff, wrd_len * 4) : qdev->read(addr, buff, wrd_len * 4);
+	qspi_cmd_sleep_rpu(&qspi_perip);
+
+#else
+	spim_cmd_rpu_wakeup_fn(0x1);
+	LOG_DBG("exited spim_cmd_rpu_wakeup_fn()\n");
+
+	spim_validate_rpu_awake_fn();
+	LOG_DBG("exited spim_validate_rpu_awake_fn(). Waiting for rpu_awake.....\n");
+
+	spim_wait_while_rpu_awake_fn();
+
+	(hl_flag) ? qdev->hl_read(addr, buff, wrd_len * 4) : qdev->read(addr, buff, wrd_len * 4);
+	spim_cmd_sleep_rpu_fn();
+
+#endif
+}
+
+static int cmd_sleep_stats(const struct shell *shell, size_t argc, char **argv)
+{
+	uint32_t addr;
+	uint32_t wrd_len;
+	uint32_t *buff;
+
+	addr = strtoul(argv[1], NULL, 0);
+	wrd_len = strtoul(argv[2], NULL, 0);
+
+	if (!validate_addr(addr, wrd_len * 4))
+		return -1;
+
+	if ((selected_blk == LMAC_ROM) || (selected_blk == UMAC_ROM)) {
+		shell_print(shell, "Error... Cannot write to ROM blocks");
+	}
+
+	buff = (uint32_t *)k_malloc(wrd_len * 4);
+
+	get_sleep_stats(addr, buff, wrd_len);
+
+	for (int i = 0; i < wrd_len; i++) {
+		LOG_DBG("0x%08x\n", buff[i]);
+	}
+
+	k_free(buff);
+	return 0;
+}
+
+/* -------------------------------------------------------------------------------------- */
+/* -------------------------------------------------------------------------------------- */
+int func_irq_config(struct gpio_callback *irq_callback_data, void (*irq_handler)())
+{
+	int ret;
+
+	ret = gpio_pin_configure_dt(&host_irq_spec, GPIO_INPUT);
+
+	ret = gpio_pin_interrupt_configure_dt(&host_irq_spec,
+			GPIO_INT_EDGE_TO_ACTIVE);
+
+	gpio_init_callback(irq_callback_data,
+			irq_handler,
+			BIT(host_irq_spec.pin));
+
+	gpio_add_callback(host_irq_spec.port, irq_callback_data);
+
+	LOG_DBG("Finished Interrupt config\n\n");
+
+	return 0;
+}
+
+int func_gpio_config(void)
+{
+	int ret;
+
+	if (!device_is_ready(iovdd_ctrl_spec.port)) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(bucken_spec.port)) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(host_irq_spec.port)) {
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure_dt(&bucken_spec,
+			GPIO_OUTPUT | (GPIO_PIN_CNF_DRIVE_H0H1 << GPIO_PIN_CNF_DRIVE_Pos));
```

Comment:
`GPIO_PIN_CNF_DRIVE_H0H1` cannot be used as a flag for this function. Include `<zephyr/dt-bindings/gpio/nordic-nrf-gpio.h>` and use `NRF_GPIO_DS_ALT` instead.